File: python_converted/src/core/services/thumbnails/ThumbnailGenerator.py
from typing import Any, List, Union


class ThumbnailOptions:
    width: float
    height: float
    quality?: float
    format?: str
    timestamp?: float
    page?: float
class ThumbnailResult:
    data: Buffer
    width: float
    height: float
    format: str
    size: float
class ThumbnailGenerator:
    generateThumbnail(input: Union[Buffer, Readable, options: ThumbnailOptions): Awaitable[ServiceResponse<ThumbnailResult>>]
    canHandle(mimeType: str): bool
    getSupportedFormats(): List[str]
    cleanup?(): Awaitable[None> 

File: src/core/services/thumbnails/ThumbnailGenerator.ts
import { Readable } from 'stream';
import { ServiceResponse } from '../types/Service';

export interface ThumbnailOptions {
  width: number;
  height: number;
  quality?: number;
  format?: string;
  timestamp?: number;  // For video thumbnails (position in seconds)
  page?: number;      // For document thumbnails
}

export interface ThumbnailResult {
  data: Buffer;
  width: number;
  height: number;
  format: string;
  size: number;
}

export interface ThumbnailGenerator {
  generateThumbnail(input: Buffer | Readable, options: ThumbnailOptions): Promise<ServiceResponse<ThumbnailResult>>;
  canHandle(mimeType: string): boolean;
  getSupportedFormats(): string[];
  cleanup?(): Promise<void>;
} 